### ISSUE TITLE
'React/*.h' file not found while building

### ISSUE THREAD

**Issue by beretboat** [ptmt/react-native-macos#178]
I see lots of header not found error while building (example project or project created by react-native-macos-cli) using the latest code. Libraries such as RCTText and RCTNetwork do not compile because their header search paths are empty in Xcode project settings. 

Is it just me?

**Comment by ptmt** [comment]
It seems that headers weren't copied into the build folder. This was introduced after RN 0.27, when React build step become a standalone one that needs to be run before main build:

![image](https://user-images.githubusercontent.com/1004115/31989814-0b18f806-b97c-11e7-95da-1d1994ff13eb.png)

I just tried react-native-macos-cli init and it worked. Can you please see you build logs checking if there is something wrong with the `React` task?

**Comment by beretboat** [comment]
In my case the headers did get copied, but the path of the headers being copied to did not included in header search paths. Actually all header search path (both user and system) are empty in all targets are empty, except for the react target in React project.

**Comment by ptmt** [comment]
As far as I understand there is no need for search paths for libraries. React needs it because it uses C++ libs headers, but when you don't need it since it's all located in the same place and you require it as `<React/>` from common `include` folder. Try to fully clean the project and ensure that React stage goes before others and all headers are copied. Is there any possibility that you get some error that prevents React target to be fully complete?

**Comment by beretboat** [comment]
React target seems fine. 

I understand that headers will be copied into "Product Directory"/include. So on my machine the headers of React.xcodeproj will be copied to "/.../MyProject/node_modules/react-native-macos/React/", and the headers of RCTImage.xcodeproj will be copied to "/.../MyProject/node_modules/react-native-macos/Libraries/Image/build/Debug/include/". 

For RCImage project, "/.../MyProject/node_modules/react-native-macos/Libraries/Image/build/Debug/include/" will be included in it's header search paths automatically, but "/.../MyProject/node_modules/react-native-macos/React/" will be not.

**Comment by ptmt** [comment]
Check build logs, Xcode allows to see them nicely:
![image](https://user-images.githubusercontent.com/1004115/32041360-71022280-ba3c-11e7-97fe-1dd5f3fb0cac.png)

As you can see, header file copied into DerivedData folder (such as `/Users/ptmt/Library/Developer/Xcode/DerivedData/test1-avttinbdmiaaocfgipwgnxdnahxr/Build/Products/Debug/include/React/NSView+React.h
`

![image](https://user-images.githubusercontent.com/1004115/32041416-9f2409f8-ba3c-11e7-9cb5-dae000f802b6.png)


No need to have search files for headers, no `node_modules` at all, all files already in `include` folder.

Can you confirm that you have that folder with all headers copied there? If you're trying to build RTCImage without prior React target, it won't build since there will be no React headers out there. I'm struggling to think of any environment differences that lead to that error. What's macOS version do you have? Can you re-create the project right now with the cli?

**Comment by beretboat** [comment]
OK I finally figure it now. There's a build location setting in Xcode. By default it looks like this:
<img width="912" alt="screen shot 2017-10-26 at 4 38 06 pm" src="https://user-images.githubusercontent.com/4593681/32042990-00cdb59e-ba6c-11e7-8973-940ecee10abd.png">
Under this setting, all headers of all targets will be copied to a single folder, the product directory of the app. 

But I changed it to "Legacy" for some reasons. Under this setting, headers will be copied to product directory of their targets. Which means the headers of React will be copied to some where like ".../node_modules/react-native-macos/React/build/", while header of RCTImage will be copied to ".../node_modules/react-native-macos/Libraries/Image/build/Debug/". That's why RCTImage could not find React headers. 

I will find a way to workaround this. Closing this issue now.

Thanks for your help!
